pkg/maps/ctmap/ctmap.go

```diff
@@ -182,6 +185,103 @@ type GCFilter struct {
 	MatchIPs map[string]struct{}
 }
 
+func CreateCtKey(remoteAddr, localAddr string, proto u8proto.U8proto, ingress bool) (CtKey, error) {
+	ip, port, err := net.SplitHostPort(remoteAddr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid remote address '%s': %s", remoteAddr, err)
+	}
+
+	sIP := net.ParseIP(ip)
+	if sIP == nil {
+		return nil, fmt.Errorf("unable to parse IP %s", ip)
+	}
+
+	sport, err := strconv.ParseUint(port, 10, 16)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse port string: %s", err)
+	}
+
+	local_ip, local_port, err := net.SplitHostPort(localAddr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid local address '%s': %s", localAddr, err)
+	}
+
+	dIP := net.ParseIP(local_ip)
+	if dIP == nil {
+		return nil, fmt.Errorf("unable to parse IP %s", local_ip)
+	}
+
+	dport, err := strconv.ParseUint(local_port, 10, 16)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse port string: %s", err)
+	}
+
+	if sIP.To4() != nil {
+		key := &CtKey4{
+			SourcePort: byteorder.HostToNetwork(uint16(sport)).(uint16),
+			DestPort:   byteorder.HostToNetwork(uint16(dport)).(uint16),
+			NextHeader: proto,
+			Flags:      TUPLE_F_OUT,
+		}
+		// CTmap has the addresses in the reverse order w.r.t. the original direction
+		copy(key.SourceAddr[:], dIP.To4())
+		copy(key.DestAddr[:], sIP.To4())
+		if ingress {
+			key.Flags = TUPLE_F_IN
+		}
+		return key, nil
+	}
+
+	key := &CtKey6{
+		SourcePort: byteorder.HostToNetwork(uint16(sport)).(uint16),
+		DestPort:   byteorder.HostToNetwork(uint16(dport)).(uint16),
+		NextHeader: proto,
+		Flags:      TUPLE_F_OUT,
+	}
+	// CTmap has the addresses in the reverse order w.r.t. the original direction
+	copy(key.SourceAddr[:], dIP.To16())
+	copy(key.DestAddr[:], sIP.To16())
+	if ingress {
+		key.Flags = TUPLE_F_IN
+	}
+	return key, nil
```

Comment:
Instead of setting the sport and dport has network byte order, you could set it as host network byte order and then here you could call `CtKey.ToNetwork()`

Reply:
Right, changed!